text/3723-static-function-argument-unpacking.md

```diff
@@ -0,0 +1,719 @@
+- [Is it possible to unpack a tuple into method arguments?](https://stackoverflow.com/questions/60381690/is-it-possible-to-unpack-a-tuple-into-method-arguments)
+
+### Unpacking Arrays
+
+Rust Internals:
+- [Pre-RFC: Array expansion syntax](https://internals.rust-lang.org/t/pre-rfc-array-expansion-syntax/13490)
+
+### Using Tuples in Place of Argument Lists
+
+Rust GitHub:
+- [Language feature: flat tuple as function arguments](https://github.com/rust-lang/rfcs/issues/2667)
+
+Rust Zulip t-lang:
+- [t-lang \> Function call tuple destructuring](https://rust-lang.zulipchat.com/#narrow/channel/213817-t-lang/topic/Function.20call.20tuple.20destructuring)
+
+Rust Users Forum:
+- [Why can’t we map tuples on arguments in Rust?](https://users.rust-lang.org/t/why-cant-we-map-tuples-on-arguments-in-rust/15628)
+
+### Unpacking Structs
+
+Rust Internals:
+- [Exploding structs](https://internals.rust-lang.org/t/exploding-structs/13884)
+
+Rust Users Forum:
+- [Unpacking struct members simultenously](https://users.rust-lang.org/t/unpacking-struct-members-simultenously/8736)
+
+### Other Related
+
+On variadic generics in general: This might in the end solve the same problem along with many others as well. Variadic generics is a more ambitious feature with a large design space, and the design progress seems to have been ongoing since 2014. Meanwhile, argument unpacking essentially provides a subset of the consequences of variadic generics designs seen so far.
+
+Rust GitHub:
+- [Draft RFC: variadic generics](https://github.com/rust-lang/rfcs/issues/376)
+- [RFC 2909: destructuring_assignment](https://github.com/rust-lang/rfcs/blob/master/text/2909-destructuring-assignment.md)
+
+Rust Internals:
+- [Variadic generics design sketch](https://internals.rust-lang.org/t/variadic-generics-design-sketch/18974)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Should argument unpacking be desugar into Alternative A or Alternative B below, or does it make any difference?
+    
+    Alternative A:
+    ```rust
+    let tup = (1, 2, 3);
+    
+    // callee_fn(...tup); desugars into:
+    {
+        let _tmp0 = tup.0;
+        let _tmp1 = tup.1;
+        let _tmp2 = tup.2;
+        callee_fn(_tmp0, _tmp1, _tmp2);
+    }
+    ```
+    
+    Alternative B:
+    ```rust
+    let tup = (1, 2, 3);
+    
+    // callee_fn(...tup); desugars into:
+    callee_fn(tup.0, tup.1, tup.2);
+    ```
+
+# Future Possibilities
+[future-possibilities]: #future-possibilities
+
+Future RFCs should freely venture outside the scope of this RFC and complement or build on this limited form of argument unpacking.
+
+## Unpacking in Fixed-Size Array and Tuple Literals
+
+The same ellipsis syntax with a very similar meaning could be adopted to defining fixed-size arrays and tuple literals as well. For example:
+```rust
+const CDE: [char; 3] = ['C', 'D', 'E'];
+const ABCDEFG1: [char; 7] = ['A', 'B', ...CDE, 'F', 'G'];
```

Comment:
If this is added, then I’d expect for the syntax to be supported in `vec!` as well.

Reply:
True. Using `vec![]` is a good example I should add to the subchapter "Unpacking Arguments for Macro Invocations" as a synergy for unpacking in arrays (or the other way around).

Reply:
I've now added a mention of `vec![]` use under "Unpacking Arguments for Macro Invocations" in commit e7f696c3f25a0d692f5f1c8a27c18e26e109c50c.